src/categorical_algebra/CSets.jl

```diff
@@ -1429,6 +1433,271 @@ end
+  elseif !isempty(state.to_recurse)
+    return Base.iterate(Sub, (state))
+  else 
+    return nothing
+  end 
+end
+
+
+# Could be cleaner/faster if we used CSetAutomorphisms to handle symmetries
+"""
+Given a list of ACSets X₁...Xₙ, find all multispans A ⇉ X ordered by decreasing 
+size of A.
+
+We search for overlaps guided by iterating through the subobjects of the 
+smallest ACSet.
+  
+If a subobject of the first object, A↪X, has multiple maps into X₁, then 
+we need to cache a lot of work because we consider each such subobject 
+independently. This is the maps from A into all the other objects as well as the 
+automorphisms of A.  
+"""
+function overlap_maps(Xs::Vector{T}) where T<:ACSet
+  !isempty(Xs) || error("Vector must not be empty")
+  Xs = Xs[sortperm(size.(Xs))] # put the smallest X first
+  res = OrderedDict()
+  for subobj in hom.(SubobjectIterator(Xs[1]))
+    abs_subobj = abstract(dom(subobj)) ⋅ subobj
+    Y = dom(abs_subobj)
+    # don't repeat work if already computed syms/maps for something iso to Y
+    seen = false
+    for (Y′, Y′maps) in collect(res)
+      σ = isomorphism(Y′, Y)
+      if !isnothing(σ)
+        push!(Y′maps[1], σ ⋅ abs_subobj)
+        seen = true
+        break
+      end
+    end
+    if seen continue end 
+    # Compute the automorphisms so that we can remove spurious symmetries
+    syms = isomorphisms(Y, Y)
+    # Get monic maps from Y into each of the objects. The first comes for free
+    maps = Vector{ACSetTransformation}[[abs_subobj]]
+    for X in Xs[2:end]
+      fs = homomorphisms(Y, X; monic=true)
+      real_fs = Set() # quotient fs via automorphisms of Y
+      for f in fs 
+        if all(rf->all(σ -> force(σ⋅f) != force(rf),  syms), real_fs)  
+          push!(real_fs, f)
+        end
+      end
+      if isempty(real_fs)
+        break # this subobject of Xs[1] does not have common overlap w/ all Xs
+      else
+        push!(maps,collect(real_fs))
+      end
+    end
+    if length(maps) == length(Xs)
+      res[Y] = maps
+    end
+  end
+  return res 
+end
+
+function partial_overlaps(Xs::Vector{T}) where T<:ACSet
+  res = []
+  for (_,v) in collect(overlap_maps(Xs))
+    append!(res, [Multispan(collect(ms)) for ms in Iterators.product(v...)])
+  end
+  return res
+end 
+partial_overlaps(Xs::ACSet...) = Xs |> collect |> partial_overlaps
+
+""" Compute the Maximimum Common C-Sets from a vector of C-Sets.
+
+Find an ACSet ```a`` with maximum possible size (``|a|``) such that there is a  
+monic span of ACSets ``a₁ ← a → a₂``. There may be many maps from this overlap 
+into each of the inputs, so a Vector{Vector{ACSetTransformations}} per overlap 
+is returned (a cartesian product can be taken of these to obtain all possible 
+multispans for that overlap). If there are multiple overlaps of equal size, 
+these are all returned.
+
+If there are attributes, we ignore these and use variables in the apex of the 
+overlap.
+"""
+function mca(Xs::Vector{T}) where T <: ACSet
```

Comment:
Can we call this `maximum_common_subobject`? It's a longer name but `mca` is cryptic.